packages/jest-resolve/src/resolver.ts

```diff
@@ -251,6 +251,9 @@ export default class Resolver {
   }
 
   isCoreModule(moduleName: string): boolean {
+    if (moduleName.startsWith('node:')) {
+      moduleName = moduleName.replace('node:', '');
```

Comment:
I would go with `.substr(5)` to ensure that only prefix has been cut.

Reply:
Thanks for the suggestion!

That should be the same, it replaces only the first occurrence, here we are checking if it starts with `node:`, so when we replace it, only the prefix has been cut.
No difference between `.substr(5)` and `.replace('node:', '')`, it is the same behavior there thanks to the if statement.

Reply:
I would say the difference is that replace does a redundant comparison first, whereas substr simply cuts the string.

Reply:
Oh yeah, I see, so `.substr(5)` should be a little bit faster.

Reply:
Usually, it is 30 times faster than `replace`.

Also, `regex` would be almost double faster than `replace`, and would take 1 line because it doesn't need an `if`.
```ts
moduleName = moduleName.replace(/^node:/, '');
```

Reply:
Right, I updated to use the regex one. :+1: 
Is it right now for you now ? :smile: 
Should be faster, it's always appreciated.

Reply:
Great, thanks! For me it was fine before too, I just added 2 cent from my side about possible options.

Reply:
Good